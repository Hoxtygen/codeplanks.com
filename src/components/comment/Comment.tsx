import moment from "moment";
import React from "react";
import { SingleComment } from "../../typedefs";
import parse from "html-react-parser";

export default function Comment({ comment }: SingleComment) {
  return (
    <div data-testid="comment" className="border-b border-gray-100 mb-4 pb-4">
      <p data-testid="commenter-name" className="mb-4">
        <span data-testid="comment-date" className="font-semibold">
          {comment.name}
        </span>{" "}
        on {moment(comment.createdAt).format("MMM DD, YYYY")}
      </p>
      <p
        data-testId="commenter-comment"
        className="whitespace-pre-line text-gray-600 w-full"
      >
        {parse(comment.comment)}
      </p>
    </div>
  );
}
