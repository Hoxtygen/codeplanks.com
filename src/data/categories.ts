export const testCategories = {
  categories: [
    {
      id: "cl8yjhbie2dj00bn1qq2wml1d",
      name: "Web Development",
      slug: "web-dev",
    },
    { id: "cl8zulkgd3hoi0bn75ccbcybd", name: "React", slug: "react" },
    {
      id: "cl8zuqc5u3k290an220qa4y4c",
      name: "Flutter",
      slug: "flutter",
    },
    { id: "cl8zuqsae3k7x0an203v29ztg", name: "NodeJS", slug: "node-js" },
  ],
};
