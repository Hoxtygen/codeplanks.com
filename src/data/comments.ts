export const testComments = {
  comments: [
    {
      id: "cl9ptdgr712um0alp2qqae0rg",
      name: "Machiavelli",
      email: "nacuabjkj dfd",
      comment:
        "Lorem ipsum dolor sit amet consectetur adipisicing elit. Quam ex commodi voluptatibus laboriosam est? Soluta, odio natus vel blanditiis mollitia impedit expedita temporibus sit cum dolores enim eveniet eaque dolorum repellat labore hic magnam non assumenda laboriosam.\n",
      createdAt: "2022-10-30T16:25:30.706593+00:00",
    },
    {
      id: "cl9ptkjjl15ob0an5upjhw2mo",
      name: "Julian Ward",
      email: "julina",
      comment:
        "Lorem ipsum dolor sit amet consectetur adipisicing elit. Quam ex commodi voluptatibus laboriosam est? Soluta, odio natus vel blanditiis mollitia impedit expedita temporibus sit cum dolores enim eveniet eaque dolorum repellat labore hic magnam non assumenda laboriosam.\n",
      createdAt: "2022-10-30T16:25:30.706593+00:00",
    },
    {
      id: "cl9v9w8zje3820an5hdkao5mu",
      name: "Marijuana Trump",
      email: "marijuana.trump@sterling.ng",
      comment:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Harum voluptates odit iste, id est aliquid molestias consequuntur nemo obcaecati aspernatur non alias doloribus laborum eveniet tempora excepturi quia in voluptatum dolor corporis adipisci odio! Architecto recusandae dolorum ullam nihil obcaecati.\n",
      createdAt: "2022-10-30T16:25:30.706593+00:00",
    },
    {
      id: "cl9wfird6s03n0blsonpo2sa6",
      name: "Job-Search",
      email: "b@example.com",
      comment: "Lorem makes sense at all times",
      createdAt: "2022-10-30T16:25:30.706593+00:00",
    },
  ],
};
