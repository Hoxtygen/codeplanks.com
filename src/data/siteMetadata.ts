const siteMetadata = {
  companyName: "Codeplanks",
  phoneNumber: "(+234) 806-018-4972",
  siteUrl: "https://codeplanks.com",
  siteLogo: "https://example.com/images/logo.png",
  siteLogoSquare: "https://example.com/images/logox200.png",
  email: "adenwas@gmail.com",
  twitter: "https://twitter.com/hoxtygen",
  twitterHandle: "@hoxtygen",
  facebook: "https://www.facebook.com/example.page",
  linkedin: "https://www.linkedin.com/in/wasiu-idowu/",
  ogImageUrl: "https://www.placholderimageurl.com",
};

export default siteMetadata;
