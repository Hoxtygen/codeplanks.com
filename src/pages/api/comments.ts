import { graphqlAPI } from "../../constants";
import { NextApiRequest, NextApiResponse } from "next";
import { GraphQLClient } from "graphql-request";
import { newCommentQuery } from "../../graphql";
import { NewComment } from "../../typedefs";

type Override<T1, T2> = Omit<T1, keyof T2> & T2;
type CustomRequest = Override<NextApiRequest, { body: NewComment }>;

export default async function formHandler(
  req: CustomRequest,
  res: NextApiResponse
) {
  const token = process.env.GRAPHCMS_TOKEN;
  if (!token) {
    throw new Error("Token missing. Please supply a token ");
  }

  const graphqlClient = new GraphQLClient(graphqlAPI, {
    headers: {
      authorization: `Bearer ${token}`,
    },
  });

  const { name, email, comment, slug } = req.body;

  const result: NextApiResponse = await graphqlClient.request(newCommentQuery, {
    name,
    email,
    comment,
    slug,
  });
  return res.status(200).send(result);
}
