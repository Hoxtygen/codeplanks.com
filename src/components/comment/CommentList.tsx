import React from "react";

import { CommentListProp, CommentProp } from "../../typedefs";
import Comment from "./Comment";

export default function CommentList({ postComments }: CommentListProp) {
  return (
    <div className="bg-white dark:bg-slate-800 bg-white shadow-lg rounded-lg p-8 pb-12 mb-8">
      <h3
        data-testid="comments-heading"
        className="text-xl mb-8 font-semibold border-b pb-4"
      >
        {postComments.length} Comments
      </h3>
      {postComments.map((comment: CommentProp) => (
        <Comment key={comment.id} comment={comment} />
      ))}
    </div>
  );
}
