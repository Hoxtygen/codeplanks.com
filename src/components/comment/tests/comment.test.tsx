import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import { testComments } from "../../../data/comments";
import Comment from "../Comment";
import CommentList from "../CommentList";

describe("Comment", () => {
  it("should render commentlist component", () => {
    render(<CommentList postComments={testComments.comments} />);

    expect(screen.getByTestId("comments-heading")).toBeInTheDocument();
    expect(screen.getByTestId("comments-heading")).toMatchSnapshot();
    expect(screen.getAllByTestId("comment")).toMatchSnapshot();
  });

  it("should render the comment component", () => {
    const comment = {
      id: "cl9ptdgr712um0alp2qqae0rg",
      name: "Machiavelli",
      email: "nacuabjkj dfd",
      comment:
        "Lorem ipsum dolor sit amet consectetur adipisicing elit. Quam ex commodi voluptatibus laboriosam est? Soluta, odio natus vel blanditiis mollitia impedit expedita temporibus sit cum dolores enim eveniet eaque dolorum repellat labore hic magnam non assumenda laboriosam.\n",
      createdAt: "2022-10-30T16:25:30.706593+00:00",
    };
    render(<Comment comment={comment} />);
    expect(screen.getByTestId("commenter-name")).toBeInTheDocument();
    expect(screen.getByTestId("commenter-comment")).toBeInTheDocument();
    expect(screen.getByTestId("commenter-name")).toMatchSnapshot();
    expect(screen.getByTestId("commenter-comment")).toMatchSnapshot();
    expect(screen.getByTestId("comment-date")).toBeInTheDocument();
    expect(screen.getByTestId("comment-date")).toMatchSnapshot();
  });
});
