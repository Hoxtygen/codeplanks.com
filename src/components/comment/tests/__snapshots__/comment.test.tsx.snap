// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment should render commentlist component 1`] = `
<h3
  class="text-xl mb-8 font-semibold border-b pb-4"
  data-testid="comments-heading"
>
  4
   Comments
</h3>
`;

exports[`Comment should render commentlist component 2`] = `
[
  <div
    class="border-b border-gray-100 mb-4 pb-4"
    data-testid="comment"
  >
    <p
      class="mb-4"
      data-testid="commenter-name"
    >
      <span
        class="font-semibold"
        data-testid="comment-date"
      >
        Machiavelli
      </span>
       
      on 
      Oct 30, 2022
    </p>
    <p
      class="whitespace-pre-line text-gray-600 w-full"
      data-testid="commenter-comment"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quam ex commodi voluptatibus laboriosam est? Soluta, odio natus vel blanditiis mollitia impedit expedita temporibus sit cum dolores enim eveniet eaque dolorum repellat labore hic magnam non assumenda laboriosam.

    </p>
  </div>,
  <div
    class="border-b border-gray-100 mb-4 pb-4"
    data-testid="comment"
  >
    <p
      class="mb-4"
      data-testid="commenter-name"
    >
      <span
        class="font-semibold"
        data-testid="comment-date"
      >
        Julian Ward
      </span>
       
      on 
      Oct 30, 2022
    </p>
    <p
      class="whitespace-pre-line text-gray-600 w-full"
      data-testid="commenter-comment"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quam ex commodi voluptatibus laboriosam est? Soluta, odio natus vel blanditiis mollitia impedit expedita temporibus sit cum dolores enim eveniet eaque dolorum repellat labore hic magnam non assumenda laboriosam.

    </p>
  </div>,
  <div
    class="border-b border-gray-100 mb-4 pb-4"
    data-testid="comment"
  >
    <p
      class="mb-4"
      data-testid="commenter-name"
    >
      <span
        class="font-semibold"
        data-testid="comment-date"
      >
        Marijuana Trump
      </span>
       
      on 
      Oct 30, 2022
    </p>
    <p
      class="whitespace-pre-line text-gray-600 w-full"
      data-testid="commenter-comment"
    >
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Harum voluptates odit iste, id est aliquid molestias consequuntur nemo obcaecati aspernatur non alias doloribus laborum eveniet tempora excepturi quia in voluptatum dolor corporis adipisci odio! Architecto recusandae dolorum ullam nihil obcaecati.

    </p>
  </div>,
  <div
    class="border-b border-gray-100 mb-4 pb-4"
    data-testid="comment"
  >
    <p
      class="mb-4"
      data-testid="commenter-name"
    >
      <span
        class="font-semibold"
        data-testid="comment-date"
      >
        Job-Search
      </span>
       
      on 
      Oct 30, 2022
    </p>
    <p
      class="whitespace-pre-line text-gray-600 w-full"
      data-testid="commenter-comment"
    >
      Lorem makes sense at all times
    </p>
  </div>,
]
`;

exports[`Comment should render the comment component 1`] = `
<p
  class="mb-4"
  data-testid="commenter-name"
>
  <span
    class="font-semibold"
    data-testid="comment-date"
  >
    Machiavelli
  </span>
   
  on 
  Oct 30, 2022
</p>
`;

exports[`Comment should render the comment component 2`] = `
<p
  class="whitespace-pre-line text-gray-600 w-full"
  data-testid="commenter-comment"
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit. Quam ex commodi voluptatibus laboriosam est? Soluta, odio natus vel blanditiis mollitia impedit expedita temporibus sit cum dolores enim eveniet eaque dolorum repellat labore hic magnam non assumenda laboriosam.

</p>
`;

exports[`Comment should render the comment component 3`] = `
<span
  class="font-semibold"
  data-testid="comment-date"
>
  Machiavelli
</span>
`;
